src/main/java/net/sf/jabref/external/push/AbstractPushToApplication.java

```diff
@@ -0,0 +1,196 @@
+
+    protected boolean couldNotCall; // Set to true in case the command could not be executed, e.g., if the file is not found
+    protected boolean couldNotConnect; // Set to true in case the tunnel to the program (if one is used) does not operate
+    protected boolean notDefined; // Set to true if the corresponding path is not defined in the preferences
+    protected JPanel settings;
+    protected final JTextField Path = new JTextField(30);
+    protected String commandPath;
+    protected String commandPathPreferenceKey;
+    protected String citeCommand = Globals.prefs.get(JabRefPreferences.CITE_COMMAND);
+    protected FormBuilder builder;
+
+
+    @Override
+    public String getName() {
+        return Localization.menuTitle("Insert selected citations into %d", getApplicationName());
+    }
+
+    @Override
+    public String getTooltip() {
+        return Localization.lang("Push to %0", getApplicationName());
+    }
+
+    @Override
+    public String getKeyStrokeName() {
+        return Localization.lang("Push to %0", getApplicationName());
+    }
+
+    @Override
+    public void pushEntries(BibtexDatabase database, BibtexEntry[] entries, String keyString, MetaData metaData) {
+
+        couldNotConnect = false;
+        couldNotCall = false;
+        notDefined = false;
+
+        initParameters();
+        commandPath = Globals.prefs.get(commandPathPreferenceKey);
+
+        // Check if a path to the command has been specified
+        if ((commandPath == null) || commandPath.trim().isEmpty()) {
+            notDefined = true;
+            return;
+        }
+
+        // Execute command
+        try {
+            Runtime.getRuntime().exec(getCommandLine(keyString));
+        }
+
+        // In case it didn't work
+        catch (IOException excep) {
+            couldNotCall = true;
+            excep.printStackTrace();
+        }
+    }
+
+    @Override
+    public void operationCompleted(BasePanel panel) {
+        if (notDefined) {
+            // @formatter:off
+            panel.output(Localization.lang("Error") + ": " 
+                    + Localization.lang("Path to %0 not defined", getApplicationName()) + ".");
+            // @formatter:on
+        } else if (couldNotCall) {
+            panel.output(getCouldNotCall());
+        } else if (couldNotConnect) {
+            panel.output(getCouldNotConnect());
+        } else {
+            panel.output(Localization.lang("Pushed citations to %0", getApplicationName()) + ".");
+        }
+    }
+
+    @Override
+    public boolean requiresBibtexKeys() {
+        return true;
+    }
+
+    /**
+     * Function to get the command to be executed for pushing keys to be cited
+     *
+     * @param keyString String containing the Bibtex keys to be pushed to the application
+     * @return String array with the command to call and its arguments
+     */
+    protected String[] getCommandLine(String keyString) {
```

Comment:
No, not really. Some PushTo use different methods to communicate. For example, for LyX/Kile a pipe/file is used to communicate. Of course, one could add the corresponding method there (with null...), but since that would not be used, this still seemed to make more sense.